### DISCUSSION TITLE
Suggestions for improved intellisense

### DISCUSSION THREAD

**Discussion by kczx3** [leafsphp/leaf#187] [Q&A]
Does anyone have a suggestion on how to improve intellisense for things we add to the Dependency Injection container?  So far the only thing I've thought to do is piggyback on the functional mode aspect and create extremely simple functions that just return the corresponding "service" and use a DocBlock to specify the return type of the function.

```php
/**
 * Get connections
 * @return \my\Connections
 */
function connections() {
    return app()->connections;
}
```

Is there any other way to do this aside from having to constantly store the dependency into a temporary variable with an @var DocBlock comment everywhere I use it?

**Comment by kczx3** [comment]
No one else has this question or perhaps at least no one has another solution?

**Comment by fadrian06** [comment]
Well what I usually do is extend the App class, register the dependencies and add the docblocks

```php
namespace YourProject;

/**
* @property-read \Your\Dependency $myDependency
*/
final class App extends \Leaf\App
{
  final public function run(
    ?callable = $callback
  ) {
    $this->register(
      'myDependency',
      function () {
        return new \Your\Dependency(/*...*/);
      }
    );

    parent::run($callback);
  }
}
```

**Reply by kczx3** [reply]
Interesting.  I hadn't thought of extending App.  This wouldn't work when using functional mode with `app()` though as that is coded to return a `\Leaf\App`.
